### ISSUE TITLE
FYI: Memory usage

### ISSUE THREAD

**Issue by jspraul** [o-oconnell/mp4grep#23]
I monitored around 13GB RAM needed to use the 1.8GB vosk-model-en-us-0.22 to process a 209MB wav file with roughly 2 hours of audio.

Until I upped the VM's RAM I couldn't get it to run without 'Segmentation fault'-ing. It didn't work well with Hyper-V's dynamic memory either, the process was always killed before memory was ballooned up to what was needed.

With the RAM pre-allocated it didn't have any problem, just took a while with no indication of progress apart from occasional 'will do an exact optimization' messages.

Thanks for the work done to create this tool!

**Comment by o-oconnell** [comment]
We have a big dependency on Vosk for the transcriptions. Models can take up much more memory at runtime than their size on disk. A quote from the documentation (https://alphacephei.com/vosk/models): "Small model typically is around 50Mb in size and requires about 300Mb of memory in runtime. Big models are for the high-accuracy transcription on the server. Big models require up to 16Gb in memory since they apply advanced AI algorithms." It's on our TODO list to improve transcriptions.

**Comment by jspraul** [comment]
I agree the smaller model would probably use less memory — I assume mp4grep processed the audio file as a stream so the duration is not an issue. I would appreciate two things:
1. Don't crash
2. Document approximate memory requirements
